[wnchan] iP

wnchan opened the pull request:
### Dukduk

> Your mind is for having ideas, not holding them.  – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Dukduk is a _personal assistant_. It helps to free your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/wnchan/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [X]  Managing tasks
- [X]  Managing deadlines and events
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

`public static void main(String[] args) {
        new Dukduk("src/main/java/data/duke.txt").run();
    }`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

wnchan committed:
Level-0, Rename, Greet, Exit

wnchan committed:
Level-1: Echo

wnchan committed:
Level-2: Add, List

wnchan committed:
Level-3: Mark as Done

wnchan committed:
Level-4: ToDo, Event, Deadline

wnchan committed:
A-TextUiTesting: Automated Text UI Testing

wnchan committed:
edit

wnchan committed:
Level-5: Handle Errors

wnchan committed:
Level-6: Delete

wnchan committed:
Level-7: Save

wnchan committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Level-7: Save

wnchan committed:
Level-8: Dates and Times

wnchan committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Level-8: Dates and Times

wnchan committed:
A-MoreOOP: Use More OOP

wnchan committed:
A-Packages: Organize into Packages

wnchan committed:
Merge remote-tracking branch 'origin/add-gradle-support'

* origin/add-gradle-support:
  Bump gradle and lib version
  Add Gradle support

juzzztinsoong commented on src/main/java/Dukduk/Parser.java:
Parser.parseTask is rather long and complicated - could be split into several functions to improve modularity and ease of adding functionality

juzzztinsoong commented on src/main/java/Dukduk/Dukduk.java:
Perhaps parsing of user input could be moved to another file instead of being in the main class

juzzztinsoong commented on src/main/java/Dukduk/Task.java:
Should the function creating a task from data string be inside task? Perhaps this function should be grouped together with the other task parsing function.

juzzztinsoong reviewed (commented):
Good job, code seems functional, although here are several changes I recommend
Similar files (Task, Deadline, Event) could be grouped together into packages - this is a rather simple task that can make project easier to navigate
Greater use of OOP - I have commented on several relevant files below.

juzzztinsoong commented on src/main/java/Dukduk/Task.java:
switch and case not indented correctly :O

juzzztinsoong commented on src/main/java/Dukduk/Parser.java:
very long conditional statement, could be better separated by line breaks for readability

juzzztinsoong reviewed (commented):
Good job, code seems functional with barely any coding violations :D, although here are several code quality changes I might recommend:
Similar files could be grouped together into packages - simple task that makes project easier to navigate
Parser.parseTask is rather long and complicated - could be split into several functions to improve modularity and ease of adding functionality
Greater use of OOP - should the function creating task from string be inside task?

wnchan committed:
Rename package name to correct format.

wnchan committed:
made changes for A-Gradle

wnchan committed:
A-JUnit: Add JUnit Tests

wnchan committed:
A-Jar: Create a JAR File

wnchan committed:
edits on junit tests

wnchan committed:
A-JavaDoc: JavaDoc

wnchan committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  A-JavaDoc: JavaDoc

remuslum commented on src/main/java/Deadline.java:
```diff
@@ -1,19 +1,24 @@
 public class Deadline extends Task {
 
-    protected String by;
+    protected LocalDateTime by;
```
You can consider a more detailed variable name such as deadlineDate to ease the understanding the meaning of the variable

remuslum commented on src/main/java/Event.java:
```diff
@@ -1,22 +1,28 @@
 
-    protected String from;
-    protected String to;
+    protected LocalDateTime fromDateTime;
```
You can consider simplifying the naming to start and end to ease understanding of what the variables mean

remuslum commented on src/main/java/Dukduk.java:
```diff
@@ -1,192 +1,81 @@
-                            throw new DukdukException("OOPS!!! Task not found. " +
-                                    "Please provide a valid task number.");
-                        }
+                        this.tasks.deleteTask(taskIndex);
```
You can consider removing the use of this.

remuslum commented on src/test/java/seedu/dukduk/DeadlineTest.java:
```diff
@@ -0,0 +1,42 @@
+
+public class DeadlineTest {
+    @Test
+    public void testToString() {
```
You can consider using a more specific function name e.g testToString_Success() to state if it is a positive or negative test.

remuslum reviewed (commented):
LGTM! Keep up the good work 😄

wnchan committed:
A-CodingStandard: Follow the Coding Standard

wnchan committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  A-CodingStandard: Follow the Coding Standard

wnchan committed:
Level-9: Find

wnchan committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Level-9: Find

wnchan committed:
A-CheckStyle: Use CheckStyle

wnchan committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  A-CheckStyle: Use CheckStyle

wnchan committed:
editted build.gradle

wnchan committed:
Level-10: GUI

wnchan committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Level-10: GUI

wnchan committed:
GUI images

wnchan committed:
Added assertions

Important files were edited to have assertions.

wnchan committed:
Merge pull request #2 from wnchan/branch-A-Assertions

Added assertions

wnchan committed:
Merge branch 'master' of https://github.com/wnchan/ip

* 'master' of https://github.com/wnchan/ip:
  Added assertions

wnchan committed:
Done refactoring on certain files

Refactoring was done on files that have too long methods
and/or nested methods.

wnchan committed:
Merge pull request #3 from wnchan/branch-A-CodeQuality

Done refactoring on certain files

wnchan committed:
Merge branch 'master' of https://github.com/wnchan/ip

* 'master' of https://github.com/wnchan/ip:
  Done refactoring on certain files

wnchan committed:
Added extension: C-BetterSearch

Before adding this extension, we can only search for tasks that match
the keyword.

Code has been edited such 'find' method now allows for partial word
matching, case-insensitive matching and searching for date/time.

wnchan committed:
Merge pull request #4 from wnchan/branch-BCD-Extension

Added extension: C-BetterSearch

wnchan committed:
Merge branch 'master' of https://github.com/wnchan/ip

* 'master' of https://github.com/wnchan/ip:
  Added extension: C-BetterSearch

wnchan committed:
A-Personality

Edited the GUI background colours, font, padding and messages that
dukduk replies with.

wnchan committed:
Edited test files due to changes in return messages.

wnchan committed:
Merge branch 'branch-A-Personality'

* branch-A-Personality:
  Edited test files due to changes in return messages.
  A-Personality

wnchan committed:
Added Ui.png to docs folder

wnchan committed:
Edited some parts to get correct title of product and fixed some bugs.

wnchan committed:
Replaced Ui.png file

wnchan committed:
Added a user guide.

wnchan committed:
Update README.md

wnchan committed:
Update README.md

wnchan committed:
Merge branch 'master' of https://github.com/wnchan/ip

* 'master' of https://github.com/wnchan/ip:
  Update README.md
  Update README.md

wnchan committed:
Fix storage method.

wnchan committed:
Update README.md

wnchan committed:
Merge branch 'master' of https://github.com/wnchan/ip

* 'master' of https://github.com/wnchan/ip:
  Update README.md

wnchan committed:
Edited files for A-MoreErrorHandling

Handled cases when mark, unmark or delete task has invalid task index.

wnchan committed:
Merge branch 'branch-A-MoreErrorHandling'

* branch-A-MoreErrorHandling:
  Edited files for A-MoreErrorHandling

wnchan committed:
Edited README.md

wnchan committed:
Added case to handle when input is 'delete' and dont have task index.

wnchan committed:
Merge branch 'branch-A-MoreErrorHandling'

* branch-A-MoreErrorHandling:
  Added case to handle when input is 'delete' and dont have task index.
